lib/order.rb

```diff
@@ -0,0 +1,66 @@
+require_relative 'texting_twilio'
+
+class Order
+
+  attr_reader :basket, :texting_service
+
+  def initialize(texting_service = TextingTwilio)
+    @basket = {}
+    @texting_service = texting_service
+  end
+
+  def add_to_basket(menu, dish_number, quantity)
+    dish = menu.select(dish_number)
+    initial_insert(dish)
+    quantity.times do
+      @basket[dish] += dish.price
+    end
+  end
+
+  def summary
+    fail "Basket empty" if empty_basket?
+    readable_basket
+  end
+
+  def submit(payment)
+    fail "Basket empty" if empty_basket?
+    fail "Payment does not match total cost" unless payment_matches_total?(payment)
```

Comment:
Metrics/LineLength: Line is too long. [83/80] (https://github.com/bbatsov/ruby-style-guide#80-character-limits)